packages/core/src/js/tracing/reactnativeprofiler.tsx

```diff
@@ -47,9 +47,17 @@ export class ReactNativeProfiler extends Profiler {
       return;
     }
 
+    const appRegistryIntegration = getAppRegistryIntegration(client);
+
+    if (!appRegistryIntegration || typeof appRegistryIntegration.onRunApplication !== 'function') {
+      // eslint-disable-next-line no-console
+      __DEV__ && console.warn('AppRegistryIntegration.onRunApplication not found or invalid.');
+      return;
```

Comment:
We should not `return`, because when tracing is disabled, we still won't the rest to execute.